core/src/main/kotlin/org/evomaster/core/output/naming/RestActionTestCaseNamingStrategy.kt

```diff
@@ -161,14 +181,52 @@ open class RestActionTestCaseNamingStrategy(
                 result.add(and)
             }
         }
+
+        val numberQueryParams = getNegativeNumberQueryParams(queryParams)
+        numberQueryParams.forEachIndexed { index, queryParam ->
+            result.add("negative")
+            result.add(queryParam.name)
+            if (index != numberQueryParams.lastIndex) {
+                result.add(and)
+            }
+        }
+
+        val emptyStringQueryParams = getEmptyStringQueryParams(queryParams)
+        emptyStringQueryParams.forEachIndexed { index, queryParam ->
+            result.add("empty")
+            result.add(queryParam.name)
+            if (index != emptyStringQueryParams.lastIndex) {
+                result.add(and)
+            }
+        }
     }
 
+    // TODO: need to check if the BooleanGene is not enclosed in an OptionalGene
```

Comment:
to handle this, you can use the check `staticCheckIfImpactPhenotype()` to see if the boolean gene is in actual use

Reply:
Hi @arcuri82! Sorry for the delay, last week I was busy working on the presentation for the thesis. All went well, so that's finished.

Regarding this, it's returning `true` for both `OptionalGene`s and `(Boolean/Integer/String)Gene`s. I assume that's actually not expected, right? Maybe it's got something to do with my unit test set up

Reply:
hi @Pgarrett phenotype is what is expressed and will impact the fitness. if an optional gene is deactivated, it is still in the genotype, but will have no impact on the phenotype. 
a gene X (regardless of its type) inside an OptionalGene that is deactivated will return false in that function, true otherwise. 
we don't want names being affected by genes that are not used. 
if this is not clear, we can have a chat about it

Reply:
Ok, isn't that taken care of by using the `RestPath.getOnlyQuery` [function](https://github.com/WebFuzzing/EvoMaster/blob/e637061636dc6d22a793b5d69cf2371e15786c18/core/src/main/kotlin/org/evomaster/core/problem/rest/RestPath.kt#L300)? Since it uses [usableQueryParamsFunction](https://github.com/WebFuzzing/EvoMaster/blob/e637061636dc6d22a793b5d69cf2371e15786c18/core/src/main/kotlin/org/evomaster/core/problem/rest/RestPath.kt#L292C17-L294) which already checks if an `OptionalGene` is active.

Reply:
Ok, then I'll add the phenotype check above, to the section in which the query params are obtained ([here](https://github.com/WebFuzzing/EvoMaster/pull/1142/files#diff-1635090843debc863feecb95a31a82cbb08c67fd5b7a66b03582062b01405d8dR168)). That way we can be certain the params being evaluated are query params that are active/have an impact on the phenotype